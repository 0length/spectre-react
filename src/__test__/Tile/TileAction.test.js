import React from 'react'
import { shallow } from 'enzyme'

import { TileAction } from '../..'

describe('TileAction', () => {
  it('Should render default .tile-action markup', () => {
    const wrapper = shallow(<TileAction />)

    expect(wrapper.type()).toBe('div')
    expect(wrapper.hasClass('tile-action')).toBe(true)
  })

  it('Should render children', () => {
    const wrapper = shallow(
      <TileAction>
        <p>hello world</p>
      </TileAction>
    )

    expect(wrapper.children().length).toBe(1)
    expect(wrapper.find('p').length).toBe(1)
  })

  it('Should pass classnames', () => {
    const wrapper = shallow(<TileAction className='extra' />)

    expect(wrapper.hasClass('extra')).toBe(true)
    expect(wrapper.hasClass('tile-action')).toBe(true)
  })

  it('Should render as a custom element', () => {
    const wrapper = shallow(<TileAction renderAs='main'>custom</TileAction>)

    expect(wrapper.text()).toBe('custom')
    expect(wrapper.hasClass('tile-action')).toBe(true)
    expect(wrapper.type()).toBe('main')
  })

  it('Should match the snapshot', () => {
    const wrapper = shallow(<TileAction />)

    expect(wrapper).toMatchSnapshot()
  })
})
