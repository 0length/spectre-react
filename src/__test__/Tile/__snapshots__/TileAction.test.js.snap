// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TileAction Should match the snapshot 1`] = `
<div
  className="tile-action"
/>
`;
